test/cql-pytest/test_select_mutation_fragments.py

```diff
@@ -0,0 +1,124 @@
+        cql.execute(f"INSERT INTO {test_table} (pk1, pk2, ck1, ck2, v) VALUES ({pk1}, {pk2}, 0, 2, 'regular val')")
+        partitions[(pk1, pk2)] = [
+                (pk1, pk2, 'memtable', 0, None, None, None, 'partition start'),
+                (pk1, pk2, 'memtable', 1, None, None, None, 'static row'),
+                (pk1, pk2, 'memtable', 2, 0   , 0   , 1   , 'range tombstone change'),
+                (pk1, pk2, 'memtable', 2, 0   , 1   , 0   , 'clustering row'),
+                (pk1, pk2, 'memtable', 2, 0   , 2   , 0   , 'clustering row'),
+                (pk1, pk2, 'memtable', 2, 0   , 3   , -1  , 'range tombstone change'),
+                (pk1, pk2, 'memtable', 3, None, None, None, 'partition end'),
+        ]
+
+    col_names = ('pk1', 'pk2', 'mutation_source_kind', 'partition_region', 'ck1', 'ck2', 'position_weight', 'mutation_fragment_kind')
+
+    def check_partition_rows(rows, expected_rows):
+        assert len(rows) == len(expected_rows)
+
+        for expected_col_values, row in zip(expected_rows, rows):
+            for col_name, col_value in zip(col_names, expected_col_values):
+                assert hasattr(row, col_name)
+                assert getattr(row, col_name) == col_value
+
+    # Point queries
+    for (pk1, pk2), expected_rows in partitions.items():
+        rows = list(cql.execute(f"SELECT * FROM MUTATION_FRAGMENTS({test_table}) WHERE pk1 = {pk1} AND pk2 = {pk2}"))
+        check_partition_rows(rows, expected_rows)
+
+    # Range scan
+    all_rows = list(cql.execute(f"SELECT * FROM MUTATION_FRAGMENTS({test_table})"))
+    for (pk1, pk2), expected_rows in partitions.items():
+        rows = [r for r in all_rows if r.pk1 == pk1 and r.pk2 == pk2]
+        check_partition_rows(rows, expected_rows)
+
+
+def test_order_by(cql, test_table):
+    """ ORDER BY is not allowed """
+    pk1 = util.unique_key_int()
+    pk2 = 0
+    cql.execute(f"INSERT INTO {test_table} (pk1, pk2, ck1, ck2, v) VALUES ({pk1}, {pk2}, 0, 0, 'vv')")
+
+    with pytest.raises(cassandra.protocol.InvalidRequest, match="ORDER BY is not supported in SELECT FROM MUTATION_FRAGMENTS\\(\\) statements"):
+        cql.execute(f"SELECT * FROM MUTATION_FRAGMENTS({test_table}) WHERE pk1 = {pk1} AND pk2 = {pk2} ORDER BY mutation_source_kind DESC")
+
+
+def test_mutation_source_kind(cql, test_table):
+    pk1 = util.unique_key_int()
+    pk2 = util.unique_key_int()
+
+    def expect_sources(*expected_sources):
+        for src in ('memtable', 'row-cache', 'sstable'):
+            rows = list(cql.execute(f"SELECT * FROM MUTATION_FRAGMENTS({test_table}) WHERE pk1 = {pk1} AND pk2 = {pk2} AND mutation_source_kind = '{src}'"))
+            if src in expected_sources:
+                assert len(rows) == 3 # partition-start, clustering-row, partition-end
+            else:
+                assert len(rows) == 0
+
+    cql.execute(f"INSERT INTO {test_table} (pk1, pk2, ck1, ck2, v) VALUES ({pk1}, {pk2}, 0, 0, 'vv')")
+    expect_sources('memtable')
```

Comment:
Can we be sure that because of some background event or the phase of the moon or something, Scylla doesn't flush the memtable without our intervention, I realize that it's very unlikely, but I would feel better if you could force Scylla *not* to flush this memtable. Do we have some sort of nodetool command or something to prevent flushing? (I know in the past we had something like this with "in-memory tables")

Reply:
I'm not aware of any mechanism that force-prevents memtable flushing. But I also don't know of any way scylla would flush memtables, other than reaching the memory limit.

Reply:
Preventing flushes = preventing new writes

Memtables can be flushed for different reasons, like running out of memtable memory, running out of commitlog space, explicit request, or moving a tablet.